backend/src/components/reports.js

```diff
@@ -41,6 +52,139 @@ async function getFacilityReports(req, res) {
+    }
+    const response = await getOperation(operation)
+    response?.value?.forEach((question) => questions.push(mapQuestionObjectForFront(question)))
+    return res.status(HttpStatus.OK).json(questions)
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+async function createSurveyResponse(req, res) {
+  try {
+    const payload = {
+      'ofm_contact@odata.bind': `/contacts(${req.body?.contactId})`,
+      'ofm_facility@odata.bind': `/accounts(${req.body?.facilityId})`,
+      'ofm_survey@odata.bind': `/ofm_surveies(${req.body?.surveyId})`,
+      'ofm_fiscal_year@odata.bind': `/ofm_fiscal_years(${req.body?.fiscalYearId})`,
+      'ofm_reporting_month@odata.bind': `/ofm_months(${req.body?.reportingMonthId})`,
+      ofm_response_type: req.body?.surveyResponseType,
+    }
+    const response = await postOperation('ofm_survey_responses', payload)
+    return res.status(HttpStatus.CREATED).json(new MappableObjectForFront(response, SurveyResponseMappings).toJSON())
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+async function getSurveyResponse(req, res) {
+  try {
+    const operation = `ofm_survey_responses(${req?.params?.surveyResponseId})`
+    const response = await getOperation(operation)
+    return res.status(HttpStatus.OK).json(new MappableObjectForFront(response, SurveyResponseMappings))
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+async function updateSurveyResponse(req, res) {
+  try {
+    const payload = new MappableObjectForBack(req.body, SurveyResponseMappings).toJSON()
+    // ofm_submitted_month field is a lookup field in CRM, so we need to replace them with data binding syntax
+    if ('_ofm_submitted_month_value' in payload) {
+      payload['ofm_submitted_month@odata.bind'] = `/ofm_months(${payload['_ofm_submitted_month_value']})`
+      delete payload['_ofm_submitted_month_value']
+    }
+    const response = await patchOperationWithObjectId('ofm_survey_responses', req.params.surveyResponseId, payload)
+    return res.status(HttpStatus.OK).json(response)
+  } catch (e) {
+    return res.status(HttpStatus.INTERNAL_SERVER_ERROR).json(e.data ? e.data : e?.status)
+  }
+}
+
+async function getQuestionResponses(req, res) {
+  try {
+    const questionResponses = []
+    let operation
+    if (req?.query?.surveyResponseId) {
```

Comment:
similar idea as in my sectionId comment: "I agree that it will throw error if the surveyResponseId is missing in the query from frontend. However, in the frontend, I've added a check to only call this api if there is a surveyResponseId. Also, I think it would be better to throw error than returning all question responses if a surveyResponseId is missing. Please let me know what you think."